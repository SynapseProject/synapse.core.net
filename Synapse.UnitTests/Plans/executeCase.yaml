Name: executeCase
Description: Tests selection of ActionGroup/Actions based on Status match to ExecuteCase
DefaultHandlerType: Synapse.Core:EmptyHandler
Actions:
- Name: ac0
  Parameters:
    Values:
      ReturnStatus: New
  ActionGroup:
    Name: ac0.g
    ExecuteCase: Any
    Parameters:
      Values:
        ReturnStatus: Running
      ForEach:
        Items:
        - Target: ReturnStatus
          Values:
          - Initializing
          - Complete
    Actions:
    - Name: ac0.g.0
      ExecuteCase: Complete
      Parameters:
        Values:
          ReturnStatus: Failed
    - Name: ac0.g.1
      ExecuteCase: Complete
      Parameters:
        Values:
          ReturnStatus: Waiting
  Actions:
  - Name: ac0.1
    ExecuteCase: Complete
    Actions:
    - Name: ac0.1.1
      Parameters:
        Values:
          ReturnStatus: Complete
      Actions:
      - Name: ac0.1.1.0
        ExecuteCase: Complete
        Parameters:
          Values:
            ReturnStatus: CompletedWithErrors
        Actions:
        - Name: ac0.1.1.0.1
          ExecuteCase: Complete
          Parameters:
            Values:
              ReturnStatus: Waiting
    - Name: ac0.1.2
      ExecuteCase: Complete
      Parameters:
        Values:
          ReturnStatus: Initializing
    - Name: ac0.1.3
      ExecuteCase: Complete
      Parameters:
        Type: Yaml
        Values:
          ReturnStatus: Complete
- Name: ac1
  ExecuteCase: Complete
  Parameters:
    Values:
      ReturnStatus: Cancelling
- Name: ac2
  Parameters:
    Values:
      ReturnStatus: CompletedWithErrors
