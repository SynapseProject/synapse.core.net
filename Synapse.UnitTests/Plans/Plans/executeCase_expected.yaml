Actions:
- Name: ac2
  Handler:
    Type: Synapse.Handlers.EmptyHandler, Synapse.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
    Config: {}
  Result:
    Status: CompletedWithErrors
    ExitData: empty
    BranchStatus: CompletedWithErrors
  Parameters:
    Values:
      ReturnStatus: CompletedWithErrors
  Actions: []
  InstanceId: 647
- Name: ac0
  Handler:
    Type: Synapse.Handlers.EmptyHandler, Synapse.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
    Config: {}
  Result:
    Status: New
    ExitData: empty
    BranchStatus: Failed
  Parameters:
    Values:
      ReturnStatus: New
  ActionGroup:
    Name: ac0.g
    ExecuteCase: Any
    Handler:
      Config: {}
    Result:
      BranchStatus: Failed
    Parameters:
      ForEach:
      - Target: ReturnStatus
        Values:
        - Initializing
        - Complete
      Values:
        ReturnStatus: Complete
    Actions:
    - Name: ac0.g
      ExecuteCase: Any
      Handler:
        Config: {}
      Result:
        Status: Initializing
        ExitData: empty
        BranchStatus: Initializing
      Parameters:
        Values:
          ReturnStatus: Initializing
      Actions: []
      InstanceId: 649
    - Name: ac0.g
      ExecuteCase: Any
      Handler:
        Config: {}
      Result:
        Status: Complete
        ExitData: empty
        BranchStatus: Failed
      Parameters:
        Values:
          ReturnStatus: Complete
      Actions:
      - Name: ac0.g.0
        ExecuteCase: Complete
        Handler:
          Type: Synapse.Handlers.EmptyHandler, Synapse.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
          Config: {}
        Result:
          Status: Failed
          ExitData: empty
          BranchStatus: Failed
        Parameters:
          Values:
            ReturnStatus: Failed
        Actions: []
        InstanceId: 651
      - Name: ac0.g.1
        ExecuteCase: Complete
        Handler:
          Type: Synapse.Handlers.EmptyHandler, Synapse.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
          Config: {}
        Result:
          Status: Waiting
          ExitData: empty
          BranchStatus: Waiting
        Parameters:
          Values:
            ReturnStatus: Waiting
        Actions: []
        InstanceId: 652
      InstanceId: 650
  Actions:
  - Name: ac0.1
    ExecuteCase: Complete
    Handler:
      Type: Synapse.Handlers.EmptyHandler, Synapse.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
      Config: {}
    Result:
      ExitData: empty
      BranchStatus: CompletedWithErrors
    Parameters: {}
    Actions:
    - Name: ac0.1.1
      Handler:
        Type: Synapse.Handlers.EmptyHandler, Synapse.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
        Config: {}
      Result:
        Status: Complete
        ExitData: empty
        BranchStatus: CompletedWithErrors
      Parameters:
        Values:
          ReturnStatus: Complete
      Actions:
      - Name: ac0.1.1.0
        ExecuteCase: Complete
        Handler:
          Type: Synapse.Handlers.EmptyHandler, Synapse.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
          Config: {}
        Result:
          Status: CompletedWithErrors
          ExitData: empty
          BranchStatus: CompletedWithErrors
        Parameters:
          Values:
            ReturnStatus: CompletedWithErrors
        Actions: []
        InstanceId: 655
      InstanceId: 654
    InstanceId: 653
  InstanceId: 648
StartInfo: {}
Result:
  Status: Failed
  BranchStatus: Failed
InstanceId: 62
